jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1614,6 +1646,9 @@ public static boolean equalsIgnoreEncodings(URI uriA, URI uriB)
         if (uriA.equals(uriB))
             return true;
 
+        if (uriA.toASCIIString().equals(uriB.toASCIIString()))
```

Comment:
when does this return true, but the condition above doesn't?

Reply:
Example: both `uriA` and `uriB` are opaque.
If `uriA` has a raw unicode, and `uriB` is utf-8 percent encoded of the same unicode.

The comparison in `URI.equals(URI)` will perform a simple string comparison against `URI.getRawSchemeSpecificPart()` of both.
This `URI.equals(URI)` comparison will fail.
But the `uriA.toASCIIString().equals(uriB.toASCIIString())` will return true.

Reply:
An example that doesn't pass `uriA.equals(uriB)`, but does pass `uriA.toASCIIString().equals(uriB.toASCIIString())`

```
uriA = {URI@2775} "jar:file:///path/to/test.jar!/bãm/"
uriB = {URI@2776} "jar:file:///path/to/test.jar!/b%C3%A3m/"
```